batch/batch/worker.py

```diff
@@ -757,8 +786,19 @@ def __init__(self):
                 log.info('cleaned up app runner')
 
     async def post_job_complete_1(self, job, run_duration):
+        status = await job.status()
+
+        if job.format_version.has_full_status_in_gcs():
+            await self.log_store.write_status_file(job.batch_id,
+                                                   job.job_id,
+                                                   job.attempt_id,
+                                                   json.dumps(status))
+
+        status.pop('container_statuses', None)
+        status.pop('error', None)
```

Comment:
Two things:

Use `del status['error']`, not pop.

I'd like it to be clearer we're not sending more stuff than we need.  Can we explicitly enumerate the fields being passed in the request?  It looks like we need: batch_id, job_id, attempt_id, state, start/end_time, (the format version?), and what's needed for the v2 db status.  In particular, I think you should batch_format_version.db_status here, and send only the fields you need.  You might make v2 format for this request look like:

```
{
  "batch_id': ...,
  ...
  "end_time": ...,
  "format_version": 2,
  "status": {
    "exit_code": ...,
    "duration": ...
  }
}
```